source/serializer.c

```diff
@@ -214,8 +215,15 @@ int aws_iot_st_msg_deserialize_from_cursor(
                 goto cleanup;
             }
             aws_byte_cursor_advance(cursor, length);
+            payload_check = 1;
         }
     }
+    if (payload_check == 0) {
+        message->payload.allocator = NULL;
+        message->payload.buffer = NULL;
+        message->payload.capacity = 0;
+        message->payload.len = 0;
```

Comment:
```
AWS_ZERO_STRUCT(*message->payload);
```

Reply:
I don't this would work because NULL and 0 are different

Reply:
I am pretty sure they are the same. But it's not a big deal.